getline.ts/src/common.ts

```diff
@@ -165,6 +164,40 @@ export class Token extends Address {
         let decimals = (await this.decimals()).toNumber();
         return internal.shift(-decimals);
     }
+
+    /**
+     * Converts this token to a PrintableToken.
+     * @returns This token as a PrintableToken.
+     */
+    public printable(): PrintableToken {
```

Comment:
Why is this part of the base class? Introduces the need to understand derived class, breaking nice hierachy order

Reply:
This is unfortunately the only easy way to do this, as Typescript does not have multiple constructors nor static constructors and apparently you can't access a protected member in a derived class constructor.

ie. this fails:

```class Foo {
    protected foo: string;
    constructor(foo: string) {
        this.foo = foo;
    }
}

class Bar extends Foo {
    constructor(foo: Foo) {
        super(foo.foo);
    }
}

error TS2446: Property 'foo' is protected and only accessible through an instance of class 'Foo'.
```

I'm not sure if this is a language bug.

Reply:
Alright, makes sense, leave it at that then